### PR TITLE
Report stats for filesystems backed by devices

### PR DESCRIPTION
**Pull request by dmuino**
Ignore 'nodev' devices (other than tmpfs)

### CODE REVIEW

File: lib/disk.cc [hunk 1]
```diff
@@ -30,11 +31,30 @@ Disk::Disk(Registry* registry, std::string path_prefix, container_handle* contai
...
+  while (fgets(line, sizeof line, fp) != nullptr) {
+    std::vector<std::string> fields;
+    split(line, &fields);
+    // if the fs line has 2 entries and the first entry is 'nodev'
+    // then the filesystem is special, and should be ignored for
+    // our reporting (with the exception of tmpfs)
+    if (fields.size() == 2 && fields[0] == "nodev") {
+      res.emplace(std::move(fields[1]));
+    }
+  }
```
Comment (sargun) [inline lib/disk.cc]: Why not do just do `res.erase("tmpfs")` here?

File: test/resources/proc/filesystems [hunk 2]
```diff
@@ -0,0 +1,33 @@
+nodev  sysfs
```
Comment (sargun) [inline test/resources/proc/filesystems]: Are these spaces, or \t?

**Comment by sargun** [comment]
left some comments. Looks fine.

File: lib/disk.cc [hunk 3]
```diff
@@ -30,11 +31,30 @@ Disk::Disk(Registry* registry, std::string path_prefix, container_handle* contai
...
+  return res;
+}
+
 // parse /proc/self/mountinfo
 std::vector<MountPoint> Disk::get_mount_points() const noexcept {
-  std::ostringstream os;
-  os << path_prefix_ << "/proc/self/mountinfo";
-  std::string file_name = os.str();
+  auto nodev_types = get_nodev_filesystems(path_prefix_);
+  auto file_name = fmt::format("{}/proc/self/mountinfo", path_prefix_);
```
[the remark this answers is not in the document]
Comment (dmuino) [reply lib/disk.cc]: It's only used for testing.

File: test/resources/proc/filesystems (hunk 2 above)
Comment (dmuino) [reply test/resources/proc/filesystems]: Good catch. They should have been \t. I'll push a fix.

File: lib/disk.cc (hunk 1 above)
Comment (dmuino) [reply lib/disk.cc]: Good point, we could remove 'tmpfs' from the unwanted file systems and avoid the comparison later on.

**Pull request merged**